AstarteDeviceSDKCSharp/Protocol/AstarteServerPropertyInterface.cs

```diff
@@ -18,18 +18,71 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+using AstarteDeviceSDK.Protocol;
 using AstarteDeviceSDKCSharp.Data;
+using AstarteDeviceSDKCSharp.Protocol.AstarteEvents;
 
 namespace AstarteDeviceSDKCSharp.Protocol
 {
-    public class AstarteServerPropertyInterface : AstartePropertyInterface
+    public class AstarteServerPropertyInterface : AstartePropertyInterface,
+    IAstarteServerValueBuilder, IAstarteServerValuePublisher
     {
         private readonly IAstartePropertyStorage _propertyStorage;
+        private readonly List<IAstartePropertyEventListener> _listeners;
 
-        public AstarteServerPropertyInterface(IAstartePropertyStorage propertyStorage) : base(propertyStorage)
+        public AstarteServerPropertyInterface(IAstartePropertyStorage propertyStorage)
+        : base(propertyStorage)
         {
             _propertyStorage = propertyStorage;
+            _listeners = new();
         }
 
+        public AstarteServerValue? Build(string interfacePath, object? serverValue,
+        DateTime timestamp)
+        {
+            AstarteInterfaceMapping? targetMapping = null;
+            AstarteServerValue? astarteServerValue = null;
+            foreach (var entry in GetMappings())
+            {
+                if (AstarteInterface.IsPathCompatibleWithMapping(interfacePath, entry.Key))
+                {
+                    targetMapping = entry.Value;
```

Comment:
Move `AstarteInterfaceMapping` statement close to the use.
```suggestion
                    AstarteInterfaceMapping targetMapping = entry.Value;
```